.pre-commit-config.yaml

```diff
@@ -0,0 +1,32 @@
+# Keep tool versions in sync with the versions in requirements-dev.txt
+minimum_pre_commit_version: 2.13.0
+default_install_hook_types: [pre-commit, commit-msg]
+default_language_version:
+    python: python3
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-toml
+      - id: check-added-large-files
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.8.4
```

Comment:
Not in sync with `requirements-dev.txt`. I suggest pinning the version in `requirements-dev.in` or just use pre-commit only.

Reply:
yap, I like having it in the requirements as the use of pre-commit is optional. I've pinned it to the same patch version and left comments as reminders to both files.